lib/oban.ex

```diff
@@ -353,6 +381,38 @@ defmodule Oban do
   """
   defdelegate config, to: Config, as: :get
 
+  @doc """
+  Synchronously execute all available jobs in a queue. All execution happens within the current
+  process and it is guaranteed not to raise an error or exit.
+
+  Draining a queue from within the current process is especially useful for testing. Jobs that are
+  enqueued by a process when Ecto is in sandbox mode are only visible to that process. Calling
+  `drain_queue/1` allows you to control when the jobs are executed and to wait synchronously for
+  all jobs to complete.
+
+  ## Failures & Retries
+
+  Draining a queue uses the same execution mechanism as regular job dispatch. That means that any
+  job failures or crashes will be captured and result in a retry. Failures are _not_ reported back
+  to the calling process. If you expect jobs to fail, would like to track failures, or need to
+  check for specific errors you can use one of these mechanisms:
+
+  * Check for side effects from job execution
+  * Use telemetry events to track success and failure
+  * Check the database for jobs with errors
+
+  ## Example
+
+  Drain a queue with three available jobs:
+
+      Oban.drain_queue(:default)
+      3
```

Comment:
Is this the number of failures, of successes, or of jobs?

Perhaps you should return a map in case you want to return more information in the future? Something like `%{total: 3}`?

Reply:
It is the number of jobs, but it is completely ambiguous. Returning a map with more information (or at least making it explicit) would be great.

Reply:
Updated to return a map with success and failure counts, i.e. `%{success: 1, failure: 0}` 👍